### PR TITLE
Added KSP module for generating fakes

### PR DESCRIPTION
**Pull request by sebvil**
As we discussed live, this module will generate a fake implementation on any `interface` annotated with `@FakeClass`. Query methods should be annotated with `@FakeQueryMethod` and command methods should be annotated with `@FakeCommandMethod`. Only the implementation for query method is usable atm.

### CODE REVIEW

**Comment by sebvil** [comment]
Current dependencies on/for this PR:
* main
  * **PR #11** <a href="https://app.graphite.dev/github/pr/sebvil/mango/11" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/sebvil/mango/11?utm_source=stack-comment).

File: app/src/main/java/com/sebastianvm/mango/database/dao/JobDao.kt [hunk 1]
```diff
@@ -4,20 +4,28 @@ import androidx.room.Dao
...
     fun getJob(id: Int): Flow<JobEntity>
 
+    @FakeQueryMethod
     @Query("SELECT * FROM JobEntity WHERE id IN (:jobIds)")
     fun loadAll(jobIds: List<Int>): Flow<List<JobEntity>>
 
+    @FakeCommandMethod
     @Insert
-    suspend fun insertAll(vararg users: JobEntity)
+    suspend fun insertAll(users: List<JobEntity>)
```
Comment (sebvil) [inline app/src/main/java/com/sebastianvm/mango/database/dao/JobDao.kt:26]: If we don't have a reason for `vararg` let's prefer `List`

**Review by amanzanero — APPROVED** [review]
l g t m

**Comment by amanzanero** [comment]
looking at the commit history I think you might need to rebase? guess it doesn't matter if you squash though

**Pull request merged**